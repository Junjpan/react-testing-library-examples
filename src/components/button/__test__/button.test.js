import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../button';
import { render, cleanup,fireEvent } from '@testing-library/React';


afterEach(cleanup) 


it('Button Render without crashing',()=>{
const div=document.createElement('div');
ReactDOM.render(<Button />,div) //this render using from ReactDOM
});

it('render button correctly',()=>{
    //using the render from react testing library
 const {getByTestId}=render(<Button label="click" />);
  expect(getByTestId('button')).toHaveTextContent('click')

})

it('calls correct function on click',()=>{
 const Hello=jest.fn();
 const {getByTestId}=render(<Button label="click" sayHello={Hello}/>);
fireEvent.click(getByTestId('button'));
expect(Hello).toHaveBeenCalled()
 

})


it('matches snapshot1',()=>{
    const {asFragment}=render(<Button label="save" />);
    expect(asFragment()).toMatchSnapshot();
})

it('matches snapshot2',()=>{
    const {asFragment}=render(<Button label="Hello" />);
    expect(asFragment()).toMatchSnapshot();
})



