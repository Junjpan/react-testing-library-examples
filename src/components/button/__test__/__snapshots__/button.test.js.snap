// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot1 1`] = `
<DocumentFragment>
  <div
    class="button-style"
    data-testid="button"
  >
    save
  </div>
</DocumentFragment>
`;

exports[`matches snapshot2 1`] = `
<DocumentFragment>
  <div
    class="button-style"
    data-testid="button"
  >
    Hello
  </div>
</DocumentFragment>
`;
